packages/sitecore-jss/src/personalization/personalization-decisions-service.ts

```diff
@@ -0,0 +1,208 @@
+import { AxiosDataFetcher } from './../axios-fetcher';
+import { HttpDataFetcher, HttpResponse } from './../data-fetcher';
+import { IncomingMessage, ServerResponse } from 'http';
+
+export interface RenderingPersonalizationDecision {
+  variantKey?: string;
+  errorMessage?: string;
+}
+
+export interface PersonalizationDecisionData {
+  renderings: {
+    [renderingId: string]: RenderingPersonalizationDecision;
+  };
+}
+
+export interface PersonalizationDecisionsService {
+  getPersonalizationDecisions(
+    routePath: string,
+    language: string,
+    renderingIds: string[],
+    req?: IncomingMessage,
+    res?: ServerResponse
+  ): Promise<PersonalizationDecisionData>;
+}
+
+export type DataFetcherResolver = <T>(config: {
+  /**
+   * The request timeout in milliseconds
+   */
+  timeout?: number;
+}) => HttpDataFetcher<T>;
+
+export type RestPersonalizationDecisionsServiceConfig = {
+  /**
+   * Hostname of decisions service; e.g. http://my.site.core; Default: '', same host as a page
+   */
+  host?: string;
+  /**
+   * Relative path from host to decisions service. Default: /sitecore/api/layout/personalization/decision
+   */
+  route?: string;
+  /**
+   * This value overrides the default service URL.
+   * Note: `host` and `route` options are ignored if `serviceUrl` is set.
+   */
+  serviceUrl?: string;
+  /**
+   * The Sitecore SSC API key your app uses
+   */
+  apiKey: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
```

Comment:
Is apiKey required?

Is siteName required?
For example, Sitecore site can be resolved via a host name.